### PR TITLE
dev/core#3498 Fix mishandled option values

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
dev/core#3498 Fix mishandled option values

Before
----------------------------------------
Options are not validated for multi-record custom import

After
----------------------------------------
Options are validated
'guessing' of fields is working better -in 5.50 looks like

image

but with this PR

image

Technical Details
----------------------------------------
The issue was the code to load the valid options for each field was not working on multi-value custom record sets 

Comments
----------------------------------------

Data patterns -do they REALLY add any value? I feel like they make it more confusing...

### CODE REVIEW

**Comment by totten** [comment]
civibot, test this please

File: tests/phpunit/CRM/Custom/Import/Parser/ApiTest.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+  /**
+   * Test the full form-flow import.
+   */
+  public function testImport(): void {
+    $this->individualCreate();
+    $this->createCustomGroupWithFieldOfType(['is_multiple' => TRUE, 'extends' => 'Contact'], 'select', 'level');
+    $customGroupID = $this->ids['CustomGroup']['level'];
+    $dateFieldID = $this->createDateCustomField(['date_format' => 'yy', 'custom_group_id' => $customGroupID])['id'];
+    $this->importCSV('custom_data_date_select.csv', [
```
Comment (totten) [inline tests/phpunit/CRM/Custom/Import/Parser/ApiTest.php:26]: There's a test failure:

```
fopen(/home/jenkins/bknix-dfl/build/core-23736-3prig/web/sites/all/modules/civicrm/tests/phpunit/CRM/Custom/Import/Parser/data/custom_data_date_select.csv): Failed to open stream: No such file or directory
```

I think maybe this file wasn't committed?

**Comment by totten** [comment]
I did some `r-run` and confirmed that the custom-data importer was better about warnings *and* more clever about matching the headers. :)

Added a comment about a test failure.  I think if that's resolved, then we can merge.

File: tests/phpunit/CRM/Custom/Import/Parser/ApiTest.php (hunk 1 above)
Comment (eileenmcnaughton) [reply tests/phpunit/CRM/Custom/Import/Parser/ApiTest.php:26]: Nice change from the extra file commits @demeritcowboy has been culling for me :-)

**Pull request merged**